Meta: Make sure that our includes are properly formatted

BenWiederhake opened the pull request:
This is about the third time we have weirdly-ordered includes, so let's add a lint that prevents these from being introduced. I don't like that we rely on weird "local include but resolve using system include paths" quirks.

While this does introduce significant overhead in terms of percentage, I think that an additional 10ms on huge commits (or less on smaller commits) are acceptable:

```
hyperfine -w1 './Meta/check-style.py AK/*.h AK/*.cpp' # Before this PR
Benchmark 1: ./Meta/check-style.py AK/*.h AK/*.cpp
  Time (mean ± σ):      20.3 ms ±   0.4 ms    [User: 17.1 ms, System: 3.5 ms]
  Range (min … max):    19.5 ms …  21.6 ms    128 runs

hyperfine -w1 './Meta/check-style.py AK/*.h AK/*.cpp' # After this PR
Benchmark 1: ./Meta/check-style.py AK/*.h AK/*.cpp
  Time (mean ± σ):      32.3 ms ±   0.4 ms    [User: 27.9 ms, System: 4.4 ms]
  Range (min … max):    31.4 ms …  34.9 ms    91 runs
```

The new code in `check-style.py` is formatted according to "black", but not the existing untouched code. It didn't seem like a good idea to reformat the entire file without any reason.

BenWiederhake commented:
CI fails only due to aarch timing out. Let's try again. (Rebased to trigger new CI.)

GMTA commented on Meta/check-style.py:
```diff
@@ -46,6 +47,16 @@
+# Make sure that all includes are either system includes or immediately resolvable local includes
+ANY_INCLUDE_PATTERN = re.compile('^ *# *include\\b.*[>"](?!\\)).*$', re.M)
+SYSTEM_INCLUDE_PATTERN = re.compile("^ *# *include *<([^>]+)>(?: //.*)?$")
+LOCAL_INCLUDE_PATTERN = re.compile('^ *# *include *"([^>]+)"(?: /[*/].*)?$')
```
Don´t you mean `[^"]` here?

Also, why do local includes allow for `/* */` type comments, but system includes do not?

BenWiederhake replied:
You're right! My intention was to remove it, but it really is used for some local includes, and forgot to allow it for system includes.

I'll change it to allow for both.

GMTA commented on Meta/check-style.py:
```diff
@@ -67,20 +78,28 @@ def find_files_here_or_argv():
     return filter(should_check_file, raw_list)
 
 
+def is_forbidden(filename, forbidden_list):
```
"forbidden" requires a bit of imagination, maybe call this something like `list_contains_filename` or `filename_matches_list`?

BenWiederhake replied:
You're right! I just copied the name from the constant without reflecting on it. I think I'll call it `is_in_prefix_list`, because the "prefix" aspect is important in my eyes.

BenWiederhake commented:
Changes since last push:
- Allow comments in system includes (thanks @GMTA!)
- Change helper function name to more descriptive `is_in_prefix_list` (thanks @GMTA!)
- Rebased for convenience
